doc/md/cli/dev.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Dev Database
+id: dev-database
+slug: /cli/dev-database
+---
+## Introduction
+
+Various Atlas commands accept the `--dev-url` flag. This option uses the [URL](url.mdx) format, and allow passing
+a development database (a twin environment) to validate schemas, simulate migration and calculate the state of the
+migration directory by replaying the historical changes. Let's go over a few examples to explain the benefits of using
+a dev/twin database.
+
+## Validation
+
+Suppose we want to the add the following `CHECK` constraint to the table below: 
+
+```hcl title="test.hcl" {6-8}
+table "t" {
+  schema = schema.test
+  column "c" {
+    type = int
+  }
+  check "ck" {
+    expr = "c <> d"
+  }
+}
+```
+
+After running [`schema apply`](reference.md#atlas-schema-apply), we get the following error, because the `CHECK`
+constraint is invalid, as column `d` does not exist.
+
+```shell
+$ atlas schema apply --url "mysql://root:pass@:3308/test" -f test.hcl
+```
+```text
+-- Planned Changes:
+-- Modify "t" table
+ALTER TABLE `test`.`t` ADD CONSTRAINT `ck` CHECK (c <> d), DROP COLUMN `c1`, ADD COLUMN `c` int NOT NULL
+✔ Apply
+Error: modify "t" table: Error 1054: Unknown column 'd' in 'check constraint ck expression'
+exit status 1
+```
+
+Atlas cannot predict such errors without applying the schema file on the database, because some cases require parsing
+and compiling SQL expressions, traverse their AST and validate them. This is already implemented by the database engine.
+
+Migration failures can leave the database in a broken state, because some like MySQL do not support transactional
```

Comment:
This is an unexpected logical jump in the text flow. Maybe numerate the 2 examples why we need the dev twin?